src/Result/inc/ResultFactory.h

```diff
@@ -23,7 +24,7 @@ namespace calculator
              * @param answer is the answer to the expression, created by the calculate
              * @param fullResult is a string stream that combines all other variables and some calculatorMessages
             */
-            Result createResult(const Expression &parsedExpression, float answer);
+            std::shared_ptr<IResult> createResult(const Expression &parsedExpression, float answer);
```

Comment:
is answer the output of the calculation? If so, you could change the result to return a pair ie. std::pair<result,answer>
that way you have it all in once location instead of having one of the arguments be an in/out plus having a return value.